backend/routes/calendar-feed.js

```diff
@@ -0,0 +1,347 @@
+    lines.push('CALSCALE:GREGORIAN');
+    lines.push('METHOD:PUBLISH');
+    lines.push(`X-WR-CALNAME:${escapeICalText(calendarName)}`);
+    lines.push('X-WR-TIMEZONE:UTC');
+
+    // Generate events for each task
+    for (const task of tasks) {
+        lines.push(generateVEvent(task, hostname));
+    }
+
+    lines.push('END:VCALENDAR');
+
+    // iCalendar requires CRLF line endings
+    return lines.join('\r\n');
+}
+
+/**
+ * GET /api/calendar/feed.ics
+ *
+ * Returns an iCalendar feed of all tasks with due dates.
+ * This route uses token-based auth (not session) for calendar app compatibility.
+ *
+ * Query Parameters:
+ *   - token (required): iCal feed token for authentication
+ *   - completed: Include completed tasks (default: false)
+ *   - project: Filter by project ID
+ *
+ * Example URL for Apple Calendar subscription:
+ *   https://your-tududi.com/api/calendar/feed.ics?token=ical_xxxxx
+ */
+router.get('/calendar/feed.ics', async (req, res) => {
+    try {
+        const { token, completed, project } = req.query;
+
+        // Validate token
+        if (!token) {
+            return res.status(401).json({
+                error: 'Authentication required',
+                message:
+                    'Please provide a calendar feed token using the ?token= parameter',
+            });
+        }
+
+        // Find user by ical_feed_token
+        const user = await User.findOne({
+            where: {
+                ical_feed_token: token,
+                ical_feed_enabled: true,
+            },
+        });
+
+        if (!user) {
+            return res.status(401).json({
+                error: 'Invalid or disabled calendar feed token',
+            });
+        }
+
+        // Build query conditions
+        const whereConditions = {
+            user_id: user.id,
+            due_date: {
+                [Op.not]: null,
+            },
+        };
+
+        // Filter out completed tasks by default
+        if (completed !== 'true' && completed !== '1') {
+            whereConditions.status = {
+                [Op.ne]: 2, // Exclude done/completed tasks
+            };
+        }
+
+        // Filter by project if specified
+        if (project) {
+            const projectId = parseInt(project, 10);
+            if (!isNaN(projectId)) {
+                whereConditions.project_id = projectId;
+            }
+        }
+
+        // Exclude recurring child tasks (they're virtual instances)
+        // We only want parent recurring tasks which will have RRULE
+        whereConditions.recurring_parent_id = null;
+
+        // Fetch tasks with associations
+        const tasks = await Task.findAll({
+            where: whereConditions,
+            include: [
+                {
+                    model: Project,
+                    as: 'Project',
+                    attributes: ['id', 'name'],
+                },
+            ],
+            order: [['due_date', 'ASC']],
+        });
```

Comment:
The idea of ​​having filters is incredible; knowing the requests and how people use Tududi, filtering by tags and area will certainly be in demand, and I think it would make sense to include it in this first version.